cocos/gfx/webgpu/webgpu-commands.ts

```diff
@@ -0,0 +1,1707 @@
+        case 'rgba16float': return Format.RGBA16F;
+        case 'r32sint': return Format.R32I;
+        case 'r32uint': return Format.R32UI;
+        case 'r32float': return Format.R32F;
+        case 'rg32sint': return Format.RG32I;
+        case 'rg32uint': return Format.RG32UI;
+        case 'rg32float': return Format.RG32F;
+        case 'rgba32sint': return Format.RGBA32I;
+        case 'rgba32uint': return Format.RGBA32UI;
+        case 'rgba32float': return Format.RGBA32F;
+        case 'rgb10a2unorm': return Format.RGB10A2;
+        case 'depth24plus': return Format.DEPTH;
+        case 'depth24plus-stencil8': return Format.DEPTH_STENCIL;
+        case 'bc1-rgba-unorm': return Format.BC1_ALPHA;
+        case 'bc1-rgba-unorm-srgb': return Format.BC1_SRGB_ALPHA;
+        case 'bc2-rgba-unorm': return Format.BC2;
+        case 'bc2-rgba-unorm-srgb': return Format.BC2_SRGB;
+        case 'bc3-rgba-unorm': return Format.BC3;
+        case 'bc3-rgba-unorm-srgb': return Format.BC3_SRGB;
+        case 'bc4-r-snorm': return Format.BC4_SNORM;
+        case 'bc6h-rgb-float': return Format.BC6H_SF16;
+        case 'bc6h-rgb-ufloat': return Format.BC6H_UF16;
+        case 'bc7-rgba-unorm': return Format.BC7;
+        case 'bc7-rgba-unorm-srgb': return Format.BC7_SRGB;
+        default:
+            return Format.BGRA8;
+    }
+};
+  
+export function WGPUFormatToGFXFormat(format: GPUTextureFormat): Format {
+    return wGPUTextureFormatToGFXFormat(format);
+}
+
+export function GFXTextureToWebGPUTexture (textureType: TextureType): GPUTextureViewDimension {
+    switch (textureType) {
+        case TextureType.TEX1D: return '1d';
+        case TextureType.TEX2D: return '2d';
+        case TextureType.TEX2D_ARRAY: return '2d-array';
+        case TextureType.TEX3D: return '3d';
+        case TextureType.CUBE: return 'cube';
+        default: {
+            console.error('Unsupported textureType, convert to WebGPUTexture failed.');
+            return '2d';
+        }
+    }
+}
+
+export function GFXTextureUsageToNative (usage: TextureUsageBit): GPUTextureUsageFlags {
+    let nativeUsage: GPUTextureUsageFlags = 0;
+    if (usage & TextureUsageBit.TRANSFER_SRC) {
+        nativeUsage |= GPUTextureUsage.COPY_SRC;
+    }
+
+    if (usage & TextureUsageBit.TRANSFER_DST) {
+        nativeUsage |= GPUTextureUsage.COPY_DST;
+    }
+
+    if (usage & TextureUsageBit.SAMPLED) {
+        nativeUsage |= GPUTextureUsage.TEXTURE_BINDING;
+    }
+
+    if (usage & TextureUsageBit.STORAGE) {
+        nativeUsage |= GPUTextureUsage.STORAGE_BINDING;
+    }
+
+    if (usage & TextureUsageBit.COLOR_ATTACHMENT || usage & TextureUsageBit.DEPTH_STENCIL_ATTACHMENT) {
+        nativeUsage |= GPUTextureUsage.RENDER_ATTACHMENT;
+    }
+
+    if (!nativeUsage) {
+        // The default value is TEXTURE_BINDING | GPUTextureUsage.COPY_DST
+        nativeUsage = GPUTextureUsage.TEXTURE_BINDING | GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT;
```

Comment:
Is `GPUTextureUsage.RENDER_ATTACHMENT` necessary?

Reply:
An error will be thrown saying "Destination texture needs to have CopyDst and RenderAttachment usage." if you use GPUTextureUsage.COPY_DST without GPUTextureUsage.RENDER_ATTACHMENT.

Reply:
I see. You might add the above message to the comment. Other developers might also wonder this.